File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        parent = {}
        val_node = {}
        pc = set()
        for p, c, isleft in descriptions:
            val_node[p] = val_node.get(p, TreeNode(p))
            val_node[c] = val_node.get(c, TreeNode(c))
            
            if(isleft):
                val_node[p].left = val_node[c]
            else:
                val_node[p].right = val_node[c]
            
            parent[c] = val_node[p]
            if(p not in parent):
                pc.add(p)
            
            pc.discard(c)
        
        
        return val_node[list(pc)[0]]

File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func createBinaryTree(descriptions [][]int) *TreeNode {
    var isParent = make(map[int]bool)
    var nodeMap = make(map[int]*TreeNode)
    for _, d := range descriptions {
        var parent, child, isLeft = d[0], d[1], d[2]
        
        if _, ok := nodeMap[child]; !ok {
            nodeMap[child] = &TreeNode{ child, nil, nil, }
        }
        
        if _, ok := nodeMap[parent]; !ok {
            nodeMap[parent] = &TreeNode{ parent, nil, nil, }
        }
        
        if isLeft == 1 {
            nodeMap[parent].Left = nodeMap[child]
        } else {
            nodeMap[parent].Right = nodeMap[child]
        }
        
        if _, ok := isParent[parent]; !ok {
            isParent[parent] = true
        }
        
        isParent[child] = false
    }
    for k, v := range isParent {
        if v == true {
            return nodeMap[k]
        }
    }
    
    return nil
}